scylla-cql/src/frame/frame_errors.rs

```diff
@@ -42,6 +44,16 @@ pub enum FrameDeserializationError {
     Lz4DecompressError(#[from] lz4_flex::block::DecompressError),
 }
 
+impl FrameDeserializationError {
+    pub fn new_header_io_error(err: std::io::Error) -> Self {
+        Self::HeaderIoError(Arc::new(err))
+    }
+
+    pub fn new_body_chunk_io_error(expected_bytes: usize, err: std::io::Error) -> Self {
+        Self::BodyChunkIoError(expected_bytes, Arc::new(err))
+    }
+}
```

Comment:
Do these have to be `pub`?

Reply:
These are used by `scylla-proxy`, but we could make an exception there and just simply construct corresponding variants directly. I'll do it, since I agree that it does not make much sense for these util functions to be public.